sdk/core/Azure.Core/src/DynamicData/AllowListConverterFactory.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core.Dynamic;
+
+namespace Azure.Core.Json
+{
+    internal class AllowListConverterFactory : JsonConverterFactory
```

Comment:
Why do we need a ConverterFactory here rather than a regular converter?

Reply:
If you can suggest a way to implement it, I'm super open to that option.  I started my investigation with that, but wasn't able to figure it out.  The issue I hit was that I couldn't find a way to add a converter that would apply to an arbitrary type.  The converter factory API lets us look at the type and decide whether or not to add a converter for it, rather than pre-loading a converter that applies to a group of types.  But if you know how, I accept suggestions!

Reply:
It would be cheaper to maintain since a converter is single type-specific (maybe polymorphic, but that's still limiting).

Reply:
Sounds like we're good to go on the factory approach, but let me know if there are any remaining concerns I missed!